### PR TITLE
Chore: use immutability check again

### PR DESCRIPTION
**Pull request by zoltanbedi**
**What this PR does / why we need it**:

In this PR I turned the immutability check back on to make sure we don't mutate the state. ~Also removed the redux logger logic as it seems that no one uses that anymore and it was for development purposes only.~ Created a separate pr for it #34220 

The question right now is that where should we fix the state mutation issues? In this pr or in next prs?

### CODE REVIEW

**Comment by hugohaggmark** [comment]
@zoltanbedi I think we should separate the removing of logActions into a separate PR. Not sure why the immutability check was removed but reading the PR `Disable these middlewares, they are too slow` doesn't sound like something we want turned on again. @dprokop @torkelo do you have more info about this?

**Review by hugohaggmark — COMMENTED** [review]
@zoltanbedi I think we should separate the removing of logActions into a separate PR. Not sure why the immutability check was removed but reading the PR `Disable these middlewares, they are too slow` doesn't sound like something we want turned on again. @dprokop @torkelo do you have more info about this?

**Comment by dprokop** [comment]
> I think we should separate the removing of logActions into a separate PR.

yes yes yes, please

> Not sure why the immutability check was removed

File: public/app/store/configureStore.ts [hunk 1]
```diff
@@ -15,23 +12,13 @@ export function addRootReducer(reducers: any) {
...
-  } as any);
-
-  const store = reduxConfigureStore<StoreState>({
+  const store = reduxConfigureStore({
     reducer: createRootReducer(),
-    middleware: [...reduxDefaultMiddleware, ...middleware] as [ThunkMiddleware<StoreState>],
+    middleware: (getDefaultMiddleware) =>
+      getDefaultMiddleware({
+        serializableCheck: false,
+        immutableCheck: { ignoredPaths: getPathsToIgnoreMutationAndSerializableCheckOn() },
```
Comment (torkelo) [inline public/app/store/configureStore.ts:20]: yea the immutableCheck is not great for performance reasons.

**Review by torkelo — COMMENTED** [review]
immutableCheck has pretty big impact on performance, and as we use redux for some high frequency actions in dashboard it is not an ideal option to use.

File: public/app/store/configureStore.ts (hunk 1 above)
Comment (zoltanbedi) [reply public/app/store/configureStore.ts:29]: can we try that again updating rtk to [1.5.1](https://github.com/reduxjs/redux-toolkit/releases/tag/v1.5.1) they seem to improved performance?

Comment (hugohaggmark) [reply public/app/store/configureStore.ts:29]: Can we do this after v8 stable?

Comment (zoltanbedi) [reply public/app/store/configureStore.ts:29]: sure. no need to rush with this.

**Comment by zoltanbedi** [comment]
It looks like that after using the 1.5.1 version this is still slow in explore.

**Pull request closed without merging**